website/docs/cloud-docs/integrations/kubernetes/index.mdx

```diff
@@ -1,68 +1,168 @@
+kind: AgentPool
+metadata:
+  name: this
+spec:
+  organization: kubernetes-operator
+  token:
+    secretKeyRef:
+      name: tfc-operator
+      key: token
+  name: agent-pool-development
+  agentTokens:
+  - name: token-red
+  agentDeployment:
+   replicas: 1
+  autoscaling:
+    targetWorkspaces:
+    - name: us-west-development
+    - id: ws-NUVHA9feCXzAmPHx
+    - wildcardName: eu-development-*
+    minReplicas: 1
+    maxReplicas: 3
+```
 
-1. Install the [Terraform Cloud Operator for Kubernetes via Helm](https://github.com/hashicorp/terraform-helm).
+</CodeBlockConfig>
 
-   ```
-   helm repo add hashicorp https://helm.releases.hashicorp.com
+In the above example, the operator ensures that at least one agent pod is continuously running and dynamically scales the number of pods up to a maximum of three based on the workload or resource demand. The operator then monitors resource demands by observing the load of the designated workspaces specified by the `name`, `id`, or `wildcardName` patterns. When the workload decreases, the operator downscales the number of agent pods.
 
-   helm install --namespace ${RELEASE_NAMESPACE} hashicorp/terraform --generate-name
-   ```
+Refer to the [agent pool API reference](/terraform/cloud-docs/integrations/kubernetes/api-reference#agentpool) for the complete `AgentPool` specification.
 
-1. To create a Terraform workspace, you can create a separate Helm chart to deploy the custom resource or examine these [examples](https://github.com/hashicorp/terraform-helm/tree/master/example).
+### Module
 
-### Upgrading
+The `Module` controller enforces an [API-driven Run workflow](/terraform/cloud-docs/run/api) and enables the deployment of Terraform modules within various workspaces as needed.
 
-When a new version of the Terraform Cloud Operator for Kubernetes Helm Chart is available from the HashiCorp Helm repository, it can be upgraded with the following command:
+The following example deploys version `1.0.0` of the `hashicorp/module/random` module in the `workspace-name` workspace.
 
+```yaml
+---
+apiVersion: app.terraform.io/v1alpha2
+kind: Module
+metadata:
+  name: my-module
+spec:
+  organization: kubernetes-operator
+  token:
+    secretKeyRef:
+      name: tfc-operator
+      key: token
+  module:
+    source: hashicorp/module/random
+    version: 1.0.0
+  workspace:
+    name: workspace-name
+  variables:
+  - name: string_length
+  outputs:
+  - name: random_string
 ```
-helm upgrade --namespace ${RELEASE_NAMESPACE} ${RELEASE_NAME} hashicorp/terraform
+
+The operator passes the workspace's `string_length` variable to the module and stores the `random_string` outputs as either a Kubernetes secret or a ConfigMap. If the workspace marks the output as "sensitive", the operator stores the `random_string` as a Kubernetes secret; otherwise, the operator stores it as a ConfigMap. The variables must be accessible within the workspace as a workspace variable, workspace variable set, or project variable set.
```

Comment:
```suggestion
The operator passes the workspace's `string_length` variable to the module and stores the `random_string` outputs as either a Kubernetes secret or a ConfigMap. If the workspace marks the output as `sensitive`, the operator stores the `random_string` as a Kubernetes secret; otherwise, the operator stores it as a ConfigMap. The variables must be accessible within the workspace as a workspace variable, workspace variable set, or project variable set.
```